x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,477 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.connector;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents a Connector in the Elasticsearch ecosystem. Connectors are used for integrating
+ * and synchronizing external data sources with Elasticsearch. Each Connector instance encapsulates
+ * various settings and state information, including:
+ * <ul>
+ *     <li>A unique identifier for distinguishing different connectors.</li>
+ *     <li>API key for authenticating with Elasticsearch, ensuring secure access.</li>
+ *     <li>A configuration mapping which holds specific settings and parameters for the connector's operation.</li>
+ *     <li>A {@link ConnectorCustomSchedule} object that defines custom scheduling.</li>
+ *     <li>A description providing an overview or purpose of the connector.</li>
+ *     <li>An error string capturing the latest error encountered during the connector's operation, if any.</li>
+ *     <li>A {@link ConnectorFeatures} object encapsulating the set of features enabled for this connector.</li>
+ *     <li>A list of {@link ConnectorFiltering} objects for applying filtering rules to the data processed by the connector.</li>
+ *     <li>The name of the Elasticsearch index where the synchronized data is stored or managed.</li>
+ *     <li>A boolean flag 'isNative' indicating whether the connector is a native Elasticsearch connector.</li>
+ *     <li>The language associated with the connector.</li>
+ *     <li>A {@link ConnectorSyncInfo} object containing synchronization state and history information.</li>
+ *     <li>The name of the connector.</li>
+ *     <li>A {@link ConnectorIngestPipeline} object specifying the data ingestion pipeline configuration.</li>
+ *     <li>A {@link ConnectorScheduling} object with the scheduling configuration to trigger data sync.</li>
+ *     <li>The type of connector.</li>
+ *     <li>A {@link ConnectorStatus} indicating the current status of the connector.</li>
+ *     <li>A sync cursor, used for incremental syncs.</li>
+ *     <li>A boolean flag 'syncNow', which, when set, triggers an immediate synchronization operation.</li>
+ * </ul>
+ */
+public class Connector implements Writeable, ToXContentObject {
+
+    private final String connectorId;
+    @Nullable
+    private final String apiKeyId;
+    @Nullable
+    private final Map<String, Object> configuration; // TODO: add explicit types
```

Comment:
Do you plan to add explicit types in a followup PR?

Reply:
Yes this will be added in a followup PR